tests/lib/util/source-code-util.js

```diff
@@ -0,0 +1,179 @@
+/**
+ * @fileoverview Tests for source-code-util.
+ * @author Ian VanSchooten
+ * @copyright 2016 Ian VanSchooten. All rights reserved.
+ * See LICENSE in root directory for full license.
+ */
+
+"use strict";
+
+//------------------------------------------------------------------------------
+// Requirements
+//------------------------------------------------------------------------------
+
+var path = require("path"),
+    fs = require("fs"),
+    os = require("os"),
+    assert = require("chai").assert,
+    sinon = require("sinon"),
+    sh = require("shelljs"),
+    assign = require("object-assign"),
+    getSourceCodeOfFiles = require("../../../lib/util/source-code-util").getSourceCodeOfFiles,
+    SourceCode = require("../../../lib/util/source-code"),
+    defaultOptions = require("../../../conf/cli-options");
+
+var originalDir = process.cwd();
+
+//------------------------------------------------------------------------------
+// Tests
+//------------------------------------------------------------------------------
+
+describe("SourceCodeUtil", function() {
+
+    var fixtureDir;
+
+    /**
+     * Returns the path inside of the fixture directory.
+     * @returns {string} The path inside the fixture directory.
+     * @private
+     */
+    function getFixturePath() {
+        var args = Array.prototype.slice.call(arguments);
+        args.unshift(fixtureDir);
+        var filepath = path.join.apply(path, args);
+        try {
+            filepath = fs.realpathSync(filepath);
+            return filepath;
+        } catch (e) {
+            return filepath;
+        }
+    }
+
+    // copy into clean area so as not to get "infected" by this project's .eslintrc files
+    before(function() {
+        fixtureDir = os.tmpdir() + "/eslint/fixtures/source-code-util";
+        sh.mkdir("-p", fixtureDir);
+        sh.cp("-r", "./tests/fixtures/source-code-util/.", fixtureDir);
+        fixtureDir = fs.realpathSync(fixtureDir);
+    });
+
+    beforeEach(function() {
+        process.chdir(fixtureDir);
```

Comment:
I'm using the exact same pattern as [cli.js](https://github.com/eslint/eslint/blob/cc08c83d0068a26e8f6c286980ef85922816d2c0/tests/lib/cli.js#L99).  Can you explain why it shouldn't be used, and what would be a better way to prevent fixture files from being ignored?